tests/plotting/test_axes.py

```diff
@@ -127,19 +177,319 @@ def test_axes_actor_shaft_type(axes_actor):
+
+    cases = get_matrix_cases()
+    angle = 42
+    origin = (1, 5, 10)
+    scale = (4, 5, 6)
+    orientation = (10, 20, 30)
+    position = (7, 8, 9)
+    user_matrix = vtkmatrix_from_array(
+        [[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [0.9, 1.0, 1.1, 1.2], [0, 0, 0, 1]]
+    )
+
+    # test each property separately and also all together
+    axes_actor._enable_orientation_workaround = True
+    if case in [cases.ALL, cases.ORIENTATION]:
+        vtk_axes_actor.SetOrientation(*orientation)
+        axes_actor.orientation = orientation
+    if case in [cases.ALL, cases.ORIGIN]:
+        vtk_axes_actor.SetOrigin(*origin)
+        axes_actor.origin = origin
+    if case in [cases.ALL, cases.SCALE]:
+        vtk_axes_actor.SetScale(*scale)
+        axes_actor.scale = scale
+    if case in [cases.ALL, cases.POSITION]:
+        vtk_axes_actor.SetPosition(*position)
+        axes_actor.position = position
+    if case in [cases.ALL, cases.ROTATE_X]:
+        vtk_axes_actor.RotateX(angle)
+        axes_actor.rotate_x(angle)
+    if case in [cases.ALL, cases.ROTATE_Y]:
+        vtk_axes_actor.RotateY(angle * 2)
+        axes_actor.rotate_y(angle * 2)
+    if case in [cases.ALL, cases.ROTATE_Z]:
+        vtk_axes_actor.RotateZ(angle * 3)
+        axes_actor.rotate_z(angle * 3)
+    if case in [cases.ALL, cases.USER_MATRIX]:
+        vtk_axes_actor.SetUserMatrix(user_matrix)
+        axes_actor.user_matrix = user_matrix
+
+    expected = array_from_vtkmatrix(vtk_axes_actor.GetMatrix())
+    actual = array_from_vtkmatrix(axes_actor.GetUserMatrix())
+    assert np.allclose(expected, actual)
+
+    default_bounds = (-1, 1, -1, 1, -1, 1)
+    assert np.allclose(vtk_axes_actor.GetBounds(), default_bounds)
+
+    # test AxesActor has non-default bounds except in ORIGIN case
+    actual_bounds = axes_actor.GetBounds()
+    if case == cases.ORIGIN:
+        assert np.allclose(actual_bounds, default_bounds)
+    else:
+        assert not np.allclose(actual_bounds, default_bounds)
+
+    # test that bounds are always default (i.e. incorrect) after disabling orientation
+    axes_actor._enable_orientation_workaround = False
+    actual_bounds = axes_actor.GetBounds()
+    assert np.allclose(actual_bounds, default_bounds)
+
+
+def test_create_axes_marker_deprecated_constructor():
+    # define valid args for PyVista version < 0.43.0
+    old_args = dict(
+        label_color='yellow',
+        x_color='red',
+        y_color='green',
+        z_color='blue',
+        xlabel='A',
+        ylabel='B',
+        zlabel='C',
+        labels_off=False,
+        line_width=20,
+        cone_radius=0.9,
+        shaft_length=0.9,
+        tip_length=0.1,
+        ambient=0.9,
+        label_size=(0.25, 0.1),
+    )
+    old_args_modified = old_args.copy()
+    with pytest.warns(
+        PyVistaDeprecationWarning,
+        match="Use of `cone_radius` is deprecated. Use `tip_radius` instead.",
+    ):
+        axes_actor = AxesActor(**old_args_modified)
```

Comment:
Lately, we have been adding error codes that work with our upgrades. This will prevent you from forgetting to remove deprecations on version upgrades.
```suggestion
        axes_actor = AxesActor(**old_args_modified)
         if pv._version.version_info >= (0, 46):
            raise RuntimeError('Remove this deprecated class')
```

Reply:
See #5120 .

Reply:
This is a good policy, thanks for highlighting it.

This should be resolved by [473fca3](
https://github.com/pyvista/pyvista/pull/5029/commits/473fca38bbe6a029037c20cc383f7cbef487380f)

Reply:
Thanks!